src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,134 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React, { PureComponent } from 'react';
+import { withSize } from '../shared/WithSize';
+import DivWithTooltip from '../shared/DivWithTooltip';
+
+import type { Thread } from '../../types/profile';
+import type { Milliseconds, StartEndRange } from '../../types/units';
+
+import './EmptyThreadIndicator.css';
+
+type SyntheticCssDeclarations = {
+  [string]: string | number,
+};
+
+type Props = {|
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: number,
+  +thread: Thread,
+  +interval: Milliseconds,
+  +unfilteredSamplesRange: StartEndRange | null,
+|};
+
+class EmptyThreadIndicator extends PureComponent<Props> {
+  _canvas: HTMLCanvasElement | null;
+  _requestedAnimationFrame: boolean | null;
+
+  _getIndicatorPositions(): {
+    startup?: SyntheticCssDeclarations,
+    shutdown?: SyntheticCssDeclarations,
+    emptyBufferStart?: SyntheticCssDeclarations,
+    emptyBufferEnd?: SyntheticCssDeclarations,
+  } {
+    const {
+      rangeStart,
+      rangeEnd,
+      width,
+      unfilteredSamplesRange,
+      interval,
+      thread: { processShutdownTime, registerTime, unregisterTime },
+    } = this.props;
+    const style = {};
+    const rangeLength = rangeEnd - rangeStart;
+    const xPixelsPerMs = width / rangeLength;
+
+    const threadEndTime =
+      unregisterTime === null ? processShutdownTime : unregisterTime;
+
+    // Did this thread startup in this time range?
+    if (registerTime > rangeStart) {
+      style.startup = {
+        left: 0,
+        width: (registerTime - rangeStart) * xPixelsPerMs,
+      };
+    }
+
+    // Did this thread shut down in this time range?
+    if (threadEndTime !== null && threadEndTime < rangeEnd) {
+      style.shutdown = {
+        right: 0,
+        width: (rangeEnd - threadEndTime) * xPixelsPerMs,
+      };
+    }
+
+    if (unfilteredSamplesRange !== null) {
+      if (
+        // Was the buffer empty at the beginning of the range?
+        unfilteredSamplesRange.start > rangeStart &&
+        // W
+        unfilteredSamplesRange.start - rangeStart > interval
```

Comment:
I believe that if this is true then the first condition is true too. (at least if `interval` is > 0). So the first condition is useless. Or I miss something ;)